### PR TITLE
New module `@turf/boolean-parallel`

### PR DESCRIPTION
**Pull request by stebogit**
Ref. https://github.com/Turfjs/turf/issues/735
```js
/**
 * Boolean-Parallel returns True if each segment of `line1` is parallel to the correspondent
 * segment of `line2`
 *
 * @name booleanParallel
 * @param {Geometry|Feature<LineString>} line1 GeoJSON Feature or Geometry
 * @param {Geometry|Feature<LineString>} line2 GeoJSON Feature or Geometry
 * @returns {Boolean} true/false if the lines are parallel
 * @example
 * var line1 = turf.lineString([[0, 0], [0, 1]]);
 * var line2 = turf.lineString([[1, 0], [1, 1]]);
 *
 * turf.booleanParallel(line1, line2);
 * //=true
 */
module.exports = function (line1, line2) {
```

The module supports only `LineStrings` (feature or geometry), leaving the user to decide how to apply the comparison among multi lines.

**Note:** the definition of comparison is based on segment having the same slope on the map plain, i.e. equal `rhumb-bearing`.
I don't know which definition has, if existing, parallelism on a sphere, but I feel it would be beyond Turf applications.

### CODE REVIEW

File: packages/turf-boolean-parallel/index.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+/**
+ * Returns Feature's type
+ *
+ * @private
+ * @param {Geometry|Feature<any>} geojson Geometry or Feature
+ * @param {string} name of the variable
+ * @returns {string} Feature's type
+ */
+function getType(geojson, name) {
```
Comment (DenisCarriere) [inline packages/turf-boolean-parallel/index.js]: @stebogit This private looks great, we should add this to [`@turf/invariant`](https://github.com/Turfjs/turf/blob/master/packages/turf-invariant).

Similar to [`getGeomType`](https://github.com/Turfjs/turf/blob/master/packages/turf-invariant/index.js#L192)

Comment (stebogit) [reply packages/turf-boolean-parallel/index.js]: Yeah, I always wondered why we didn't have that there :smile:

Comment (DenisCarriere) [reply packages/turf-boolean-parallel/index.js]: 👍 I'll add it

File: packages/turf-boolean-parallel/test.js [hunk 2]
```diff
@@ -13,7 +13,6 @@ test('turf-boolean-parallel', t => {
         const [line1, line2] = geojson.features;
         const result = booleanParallel(line1, line2);
 
-        if (process.env.SHAPELY) shapely.contains(line1, line2).then(result => t.true(result, '[true] shapely - ' + name));
```
Comment (stebogit) [inline packages/turf-boolean-parallel/test.js]: @DenisCarriere why did you remove this?
Many other boolean modules have that, some of them also have
```js
if (process.env.JSTS) t.false(booleanJSTS('within', feature1, feature2), '[false] JSTS - ' + name);
```
What's their use exactly?

**Comment by stebogit** [comment]
Woo-hoo!! 😄:+1:
